File: app/src/main/java/com/arcgistest/activities/launcher/LauncherActivity.java
package com.arcgistest.activities.launcher;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.arcgistest.R;
import com.arcgistest.activities.main.MainActivity;

public class LauncherActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        this.setContentView(R.layout.activity_launcher);

        initData();
    }

    private void initData() {
        actionOpenMainActivity();
    }

    private void actionOpenMainActivity() {
        this.overridePendingTransition(android.R.anim.fade_in, android.R.anim.fade_out);
        Intent i = new Intent(this, MainActivity.class);
        this.startActivity(i);
        this.overridePendingTransition(android.R.anim.fade_in, android.R.anim.fade_out);
        this.finish();
    }
}

File: app/src/main/java/com/arcgistest/activities/launcher/LauncherActivity.kt
package com.arcgistest.activities.launcher

import android.os.Bundle
import com.arcgistest.R
import com.arcgistest.activities.base.BaseActivity
import com.arcgistest.activities.launcher.di.DaggerLauncherComponent
import com.arcgistest.activities.launcher.di.LauncherComponent
import com.arcgistest.activities.launcher.di.LauncherModule
import javax.inject.Inject

class LauncherActivity : BaseActivity(), LauncherView {

    companion object {
        val TAG: String = LauncherActivity::class.java.simpleName
    }

    val component: LauncherComponent by lazy {
        DaggerLauncherComponent.builder()
                .parent(appComponent)
                .module(LauncherModule())
                .target(this)
                .build()
    }

    @Inject
    lateinit var presenter: LauncherPresenter

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_launcher)
        component.inject(this)
        presenter.bindView(this)
        presenter.showMainView();
    }

    override fun onDestroy() {
        super.onDestroy()
        presenter.unbindView()
    }
}